File: Data Structures & Algorithms/is-anagram/submission-3.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }

        vector<int> charMap(26, 0);

        for (int i = 0; i < s.length(); i++) {
            charMap[s[i]-'a']++;
            charMap[t[i] - 'a']--;
        }

        for (int num : charMap) {
            if (num != 0) {
                return false;
            }
        }
        
        return true;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-3.py
def charMap(s: str) -> List[int]:
        chars = [0]*26
        for char in s:
            chars[ord(char)-ord('a')] += 1
        return chars

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramMaps = {}
        for s in strs:
            stringMap = charMap(s)
            if str(stringMap) not in anagramMaps:
                anagramMaps[str(stringMap)] = [s]
            else:
                anagramMaps[str(stringMap)].append(s)
        return list(anagramMaps.values())